modules/siddhi-core/src/main/java/io/siddhi/core/event/stream/holder/SnapshotableStreamEventQueue.java

```diff
@@ -47,6 +47,11 @@
     private boolean forceFullSnapshot = true;
     private boolean isOperationLogEnabled = true;
     private int eventIndex = -1;
+    protected int size;
+
+    public void reduceSizeByOne() {
```

Comment:
Why this is public?

Reply:
https://github.com/niruhan/siddhi/blob/53466a751c2bea850aa9fed867fd9e61d903f2c9/modules/siddhi-core/src/main/java/io/siddhi/core/util/collection/operator/SnapshotableEventQueueOperator.java#L93 gets called from snapshotableeventqueueoperator

Reply:
shoudn't it be called inside remove()?
storeEventQueue.remove();